bin/main.ml

```diff
@@ -61,17 +61,33 @@ module Prompt = struct
     | true -> Lwt_io.printl "I: reading from stdin. Please type the secret and then do Ctrl+d twice to terminate input"
     | false -> Lwt.return_unit
 
-  let read_input_from_stdin () = Lwt_io.(read stdin)
-
-  let rec input_and_validate_loop ?(transform = fun x -> x) get_secret_input =
-    let%lwt input = get_secret_input () in
-    let secret = transform input in
+  let read_input_from_stdin ?initial:_ () = Lwt_io.(read stdin)
+
+  let rec input_and_validate_loop ?(transform = fun x -> x) ?initial get_secret_input =
+    let remove_trailing_newlines s =
+      let rec loop s =
+        match String.ends_with ~suffix:"\n" s with
+        | true -> loop (String.sub s 0 (String.length s - 1))
+        | false -> s
+      in
+      loop s
+    in
```

Comment:
Is there any point in supporting windows' `\n\r` EOL? I don't think so because the rest of the code seem to make some assumption about being somewhat unixy. Maybe we should mark `available: os != "win32"` in the opam file?

also NIT: That's somewhat heavy although in practice it's unlikely to have more than a couple loops of recursions so I think it's ok

Reply:
> Maybe we should mark available: os != "win32" in the opam file?
I think that's a good idea. passage isn't running in windows now and won't be in the future either. Will push this change to master.